CesiumGltf/include/CesiumGltf/TextureView.h

```diff
@@ -0,0 +1,224 @@
+#pragma once
+
+#include "CesiumGltf/ImageCesium.h"
+#include "CesiumGltf/KhrTextureTransform.h"
+#include "CesiumGltf/Sampler.h"
+#include "CesiumGltf/TextureInfo.h"
+
+#include <vector>
+
+namespace CesiumGltf {
+
+struct Model;
+
+/**
+ * @brief Describes options for constructing a view on a glTF texture.
+ */
+struct TextureViewOptions {
+  /**
+   * @brief Whether to automatically apply the `KHR_texture_transform` extension
+   * to the texture view, if it exists.
+   *
+   * A glTF texture may contain the `KHR_texture_transform` extension, which
+   * transforms the texture coordinates used to sample the texture. The
+   * extension may also override the TEXCOORD set index that was specified by
+   * the original texture info.
+   *
+   * If a view is constructed with applyKhrTextureTransformExtension set to
+   * true, it should automatically apply the texture transform to any UV
+   * coordinates used to sample the texture. If the extension defines its own
+   * TEXCOORD set index, it will override the original value.
+   *
+   * Otherwise, if the flag is set to false, UVs will not be transformed and
+   * the original TEXCOORD set index will be preserved. The extension's values
+   * may still be retrieved using getTextureTransform, if desired.
+   */
+  bool applyKhrTextureTransformExtension;
+
+  /**
+   * @brief Whether to copy the input image.
+   *
+   * By default, a view is constructed on the input glTF image without copying
+   * its pixels. This can be problematic for clients that move or delete the
+   * original glTF model. When this flag is true, the view will manage its own
+   * copy of the pixel data to avoid such issues.
+   */
+  bool makeImageCopy;
+
+  TextureViewOptions()
+      : applyKhrTextureTransformExtension(false), makeImageCopy(false) {}
```

Comment:
You could leave this out, and instead specify the default values inline. In addition to just being less code, that would allow `TextureViewOptions` fields to be initialized with brace initialization.